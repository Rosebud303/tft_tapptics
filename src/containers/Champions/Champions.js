import React, { Component } from 'react';
import { connect } from 'react-redux';
import Modal from 'react-modal';


class Champions extends Component {
  constructor() {
    super()
    this.state = {
      showModal: false,
      currentChampion: { attributes: { data: {} } }
    }
  }

  handleOpenModal = (e) => {
    const { id } = e.target
    let champion = this.props.champions.find(champion => champion.id == id);
    this.setState({currentChampion: champion, showModal: true})
  }

  handleCloseModal = () => {
    this.setState({showModal: false, currentChampion: { attributes: { data: {} } }});
  }
  

  render() {
    const displayChampions = this.props.champions.map(champ => (
        <article id={champ.id} onClick={(e) => this.handleOpenModal(e)}>
          <img id={champ.id} src={champ.attributes.data.champion_thumbnail}/>
        </article>))

    let displayModal = (
      <article>
        <img src={this.state.currentChampion.attributes.data.champion_thumbnail} />
        <h3>{this.state.currentChampion.attributes.data.name}</h3>
        <button className="update-modal" onClick={this.handleCloseModal}>X</button>
      </article>)
  
    return (
      <section className="Champions">
        <h3>All Champions</h3>
        {displayChampions}
        <Modal 
           isOpen={this.state.showModal}
           contentLabel="Champion Modal">
           <div className="modal-display-champions">
          { this.state.showModal && displayModal }
          </div>
        </Modal>
      </section>
    )
  }
}

export const mapStateToProps = state => ({
  champions: state.champions
})

export default connect(mapStateToProps)(Champions)