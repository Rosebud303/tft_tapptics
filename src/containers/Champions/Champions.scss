.Champions {
  background-image: linear-gradient($fade1, $fade2), url('../../assets/images/champs.jpg');
  background-size: cover;
  background-repeat: no-repeat;
  height: 90vh;
  padding: $page-padding;

  h3 {
    color: $body-font-color;
    font-size: 1.5rem;
    margin: 0;
  }

  .display-champs {
    display: flex;
    flex-wrap: wrap;
    justify-content: space-between;
    text-align: left;
    overflow: scroll;
    height: 80vh;

    article {
      display: inline;
      margin: .75rem;
    }
  }
}

.Modal {
  background: $medium-bg;
  box-shadow: 2px 2px 2px darken($dark-bg, 20%);
  color: $btn-bg;
  height: 55%;
  margin: auto;
  padding: 1.5rem;
  position: relative;
  bottom: -20%;
  width: 65%;

  button {
    background-color: $dark-bg;
    border: 1px solid $btn-border;
    border-radius: 25px;
    color: $btn-bg;
    font-size: 1.25rem;
    grid-area: x;
    height: 3rem;
    width: 3rem;
  }

  .modal-display-champions {
    height: 50%;
  }

  .display-modal {
    display: grid;
    grid-template-areas:
    "img name cost x"
    "img class range ."
    "health stats stats ."
    "p-img p-info p-info .";
    grid-template-rows: .5fr .5fr .5fr 1.5fr;
    height: 200%;
    overflow: scroll;
  
    .modal-range {
      grid-area: range;
    }

    .modal-name {
      grid-area: name;
    }

    .modal-class-origin {
      grid-area: class;
      margin: auto;
    }
  
    .modal-cost {
      grid-area: cost;
      margin: auto;
    }

    .modal-health {
      grid-area: health;
    }

    .modal-stats {
      grid-area: stats;
      display: inline-flex;
      flex-wrap: wrap;
      justify-content: space-around;
    }
  
    .modal-thumbnail {
      grid-area: img;
      margin: auto;
    }

    .modal-p-thumbnail {
      grid-area: p-img;
      margin: auto;
    }

    .modal-p-name {
      grid-area: p-name;
    }

    .modal-p-info {
      grid-area: p-info;
    }

    h5 {
      font-weight: 400;
    }
  }
}
