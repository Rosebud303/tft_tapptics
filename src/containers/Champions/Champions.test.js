import React from 'react';
import { shallow } from 'enzyme';
import { Champions, mapStateToProps } from './Champions';

describe('Champions', () => {

  let wrapper;
  let mockChampions = [{
    id: 1,
    attributes: {
    data: {
      health: [],
      ability_info: {
        attributes: [{damage: [] }],
      }
    },
    origin_class_type: {
    data: [{
      attributes: {
        data: {
    tier_info: [], tiers: [],
        }
      }
    }]
  }
}}]

  beforeEach(() => {
    wrapper = shallow(<Champions champions={mockChampions} />);
  });

  it('should match the snapshot with all the data passed in', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should handle open modals', () => {
    wrapper.setState({openModal: false});
    const e = { target: { id: 1 } }
    wrapper.instance().handleOpenModal(e);
    expect(wrapper.state('showModal')).toEqual(true);
  });

  it('should handle closed modals', () => {
    wrapper.setState({openModal: true});
    wrapper.instance().handleCloseModal();
    expect(wrapper.state('showModal')).toEqual(false);
  });

});

describe('mapStateToProps', () => {

  it('should return an object with the champions array', () => {
    const mockState = {
      champions: [{name: 'Varus'}]
    }
    const expected = {
      champions: [{name: 'Varus'}]
    }

    const mappedProps = mapStateToProps(mockState);
    expect(mappedProps).toEqual(expected)
  })
});