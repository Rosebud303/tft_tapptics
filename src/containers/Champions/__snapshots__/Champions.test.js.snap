// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Champions should match the snapshot with all the data passed in 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;
